internal/pkg/vm/internal/vmcontext/invocation_context.go

```diff
@@ -191,6 +191,17 @@ func (ctx *invocationContext) invoke() (ret returnWrapper, errcode exitcode.Exit
 	// 3. transfer funds carried by the msg
 	ctx.rt.transfer(ctx.msg.from, ctx.msg.to, ctx.msg.value)
 
+	var found bool
+	ctx.toActor, found, err = ctx.rt.state.GetActor(ctx.rt.context, ctx.msg.to)
+	if !found {
+		runtime.Abort(exitcode.ErrNotFound)
+	}
+
+	ctx.fromActor, found, err = ctx.rt.state.GetActor(ctx.rt.context, ctx.msg.from)
+	if !found {
+		runtime.Abort(exitcode.ErrNotFound)
+	}
```

Comment:
Both the actor states are loaded in the `transfer` call. Return them from that instead of loading them again here.

Reply:
They are loaded iff transfer is called with a non-zero amount, else it bails:
https://github.com/filecoin-project/go-filecoin/blob/ecfbd2bf822f835c825809fadec7c04a0f13a013/internal/pkg/vm/internal/vmcontext/vmcontext.go#L525-L534

Reply:
couple ideas:
1. Load them before the `if amount.Nil() || amount.IsZero()` check. seems equally as bad
2. Return a boolean in addition to the actors indicating if state changed (ie transfer amount was non-zero)
3. Larger refactor than is here

Reply:
Ok lets do the check outside of transfer. One of the three call sites already does this, and the other one will never transfer zero.